src/test/java/org/springframework/mock/http/client/MockClientHttpResponse.java

```diff
@@ -0,0 +1,76 @@
+package org.springframework.mock.http.client;
+
+/*
+ * ⁣​
+ * Riptide
+ * ⁣⁣
+ * Copyright (C) 2015 - 2016 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.mock.http.MockHttpInputMessage;
+import org.springframework.util.Assert;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class MockClientHttpResponse extends MockHttpInputMessage implements ClientHttpResponse {
+
+    private final HttpStatus status;
+
+    /**
+     * Constructor with response body as a byte array.
+     */
+    public MockClientHttpResponse(byte[] body, HttpStatus statusCode) {
+        super(body);
+        Assert.notNull(statusCode, "statisCode is required");
+        this.status = statusCode;
+    }
+
+    /**
+     * Constructor with response body as InputStream.
+     */
+    public MockClientHttpResponse(InputStream body, HttpStatus statusCode) {
+        super(body);
+        Assert.notNull(statusCode, "statisCode is required");
+        this.status = statusCode;
+    }
+
+    @Override
+    public HttpStatus getStatusCode() throws IOException {
+        return this.status;
+    }
+
+    @Override
+    public int getRawStatusCode() throws IOException {
+        return this.status.value();
+    }
+
+    @Override
+    public String getStatusText() throws IOException {
+        return this.status.getReasonPhrase();
+    }
+
+    @Override
+    public void close() {
+        try {
+            this.getBody().close();
+        } catch (IOException e) {
```

Comment:
It's more or less the same that the real `HttpClientResponse` implementations are doing. I feel dirty for classpath-shadowing this class, but the behaviour is closer to the real world.